README.md

```diff
@@ -15,34 +18,144 @@ pitfalls.
+| <code>--warning_level (-W) [QUIET &#124; DEFAULT &#124; VERBOSE]</code>  | Specifies the warning level to use.  |
+
+#### See the [Google Developers Site](https://developers.google.com/closure/compiler/docs/gettingstarted_app) for documentation including instructions for running the compiler from the command line.
+
+### NodeJS API
+You can access the compiler in a JS program by importing
+`google-closure-compiler`:
+
+```javascript
+import closureCompiler from 'google-closure-compiler';
+const { compiler } = closureCompiler;
+
+new compiler({
+  js: 'file-one.js',
+  compilation_level: 'ADVANCED'
+});
+```
+
+This package will provide programmatic access to the native Graal binary in most
+cases, and will fall back to the Java version otherwise.
+
+#### Please see the [closure-compiler-npm](https://github.com/google/closure-compiler-npm/tree/master/packages/google-closure-compiler) repository for documentation on accessing the compiler in JS.
+
+## Compiling Multiple Scripts
+
+If you have multiple scripts, you should compile them all together with one
+compile command.
+
+```bash
+google-closure-compiler in1.js in2.js in3.js --js_output_file out.js
+```
+
+You can also use minimatch-style globs.
+
+```bash
+# Recursively include all js files in subdirs
+google-closure-compiler 'src/**.js' --js_output_file out.js
+
+# Recursively include all js files in subdirs, excluding test files.
+# Use single-quotes, so that bash doesn't try to expand the '!'
+google-closure-compiler 'src/**.js' '!**_test.js' --js_output_file out.js
+```
+
+The Closure Compiler will concatenate the files in the order they're passed at
+the command line.
+
+If you're using globs or many files, you may start to run into problems with
+managing dependencies between scripts. In this case, you should use the [Closure
+Library](https://developers.google.com/closure/library/). It contains functions
+for enforcing dependencies between scripts, and Closure Compiler will re-order
+the inputs automatically.
+
+## Getting Help
+
+1.  Post in the [Closure Compiler Discuss
+    Group](https://groups.google.com/forum/#!forum/closure-compiler-discuss).
+1.  Ask a question on [Stack
+    Overflow](https://stackoverflow.com/questions/tagged/google-closure-compiler).
 1.  Consult the [FAQ](https://github.com/google/closure-compiler/wiki/FAQ).
 
-## Building it Yourself
+## Building
 
-Note: The Closure Compiler requires [Java 8 or higher](https://www.java.com/).
+To build the compiler yourself, you will need the following:
 
-### Using [Bazel](https://bazel.build/)
+| Prerequisite | Description |
+| --- | --- |
+| [Java 8 or later](https://java.com) | Used to compile the compiler's source code. |
+| [Git](https://git-scm.com/) | Used by Bazel to download dependencies. |
+| [Bazel](https://docs.bazel.build/versions/master/install.html) | Used to build the various compiler targets. |
 
-Building using bazel requires `git` (to download dependencies).
+### Installing Bazel
```

Comment:
In general users should prefer bazelisk. Otherwise they'll get errors as the default bazel version changes but the BUILD files are not.

Reply:
I didn't see any references to Bazelisk in the original README. I wonder if @lauraharker has thoughts on this. Happy to make this change, just looking for input.

Reply:
For clarity, Bazelisk is a small wrapper around all the versions of Bazel. It reads the `.bazelversion` file and then delegates to the specified version.

My concern is that any commands we mention should call bazelisk, not bazel. This probably also includes commands in the package JSON.

If we don't, users will get difficult to debug errors. The BUILD files will parse with many versions of Bazel, but the behaviour will be slightly different and in many cases won't build. The diagnostics when this happens point mention missing env variables or build targets; nothing to do with the real problem.

I don't think we need to belabor the distinction between Bazel and Bazelisk. We just want to point the most people to the best chance of success. And possibly convey that they are approximately the same thing.

Reply:
But if *I* ran `bazelisk` right now, it would fail. I am morally opposed to catering to Windows users specifically for command-line examples - would adding a note ("on Windows, replace `bazel` with `bazelisk`") or something similar work?

Reply:
I think there was a mistake last time I edited this file. bazel nor bazelisk is installed by default on any OS that I'm aware of. Additionally, bazelisk should be used regardless of OS.

Reply:
Oh, very interesting, so this is a substantive change to the installation instructions. I can make that change for you if you summarize what should be swapped out, or you can do it directly.